src/hipscat_import/index/map_reduce.py

```diff
@@ -31,16 +31,44 @@ def create_index(args):
     )
 
     if args.include_order_pixel:
-        ## Take out the hive dictionary behavior.
+        ## Take out the hive dictionary behavior that turns these into int32.
         data["Norder"] = data["Norder"].astype(np.uint8)
         data["Dir"] = data["Dir"].astype(np.uint64)
         data["Npix"] = data["Npix"].astype(np.uint64)
+
+    # There are some silly dask things happening here:
+    # - Turn the existing index column into a regular column
+    # - If that had been the _hipscat_index, and we don't want it anymore, drop it
+    # - Create a new index, using our target indexing_column.
+    #   Use division hints if provided.
     data = data.reset_index()
     if not args.include_hipscat_index:
         data = data.drop(columns=[HIPSCAT_ID_COLUMN])
-    data = data.drop_duplicates()
+
+    if args.divisions is not None and len(args.divisions) > 2:
+        data = data.set_index(args.indexing_column, divisions=args.divisions)
+    else:
+        data = data.set_index(args.indexing_column)
+
+    if args.drop_duplicates:
+        # More dask things:
+        # - Repartition the whole dataset to account for limited memory in
+        #   pyarrow in the drop_duplicates implementation (
+        #   "array cannot contain more than 2147483646 bytes")
+        # - Reset the index, so the indexing_column values can be considered
+        #   when de-duping.
+        # - Drop duplicate rows
+        # - Set the index back to our indexing_column, but this time, the
+        #   values are still sorted so it's cheaper.
+        data = (
+            data.repartition(partition_size=1_000_000_000)
+            .reset_index()
```

Comment:
These reset_index->set_index operations seem expensive generally, though it's nice you're able to set the sorted flag. It seems like drop_duplicates doesn't have a way to consider the index, which makes sense why we have to do this. I wonder if there's some way to use groupby to get the duplicated indices, and apply a function that dedupes from each group. It may be a toss-up whether having a groupby or a set_index is more performance intensive though...

Reply:
This might be very wrong, I think I have some questions about duplicates

Reply:
After spending an hour on this, I've decided it's a dask rabbit-hole. Group by and other aggregators are subtly different between pandas and the dask dataframe implementation (often with totally good reason). If it's alright, I'm going to open an issue to investigate further (that someone else is welcome to take on!) so we can move forward with the other incremental improvements from this PR.

https://github.com/astronomy-commons/hipscat-import/issues/206